internal/controller/backup_controller.go

```diff
@@ -366,6 +369,10 @@ func (r *BackupReconciler) reconcileSnapshotBackup(
 		return &ctrl.Result{RequeueAfter: 10 * time.Second}, nil
 	}
 
+	if len(targetPod.Status.ContainerStatuses) == 0 {
+		return nil, fmt.Errorf("pod container statuses list is empty")
+	}
```

Comment:
here the logic does what we want. The overall `reconcileSnapshotBackup` fails with error,
and then the reconciliation gets re-looped.